src/mysky/index.ts

```diff
@@ -161,7 +204,13 @@ export class MySky {
       permissions.push(perm1, perm2, perm3);
     }
 
-    MySky.instance = new MySky(connector, permissions, hostDomain);
+    const currentPortalUrl = await currentUrlClient.portalUrl();
+    MySky.instance = new MySky(connector, permissions, hostDomain, currentPortalUrl);
```

Comment:
> Is is possible to call this new more than once?

Nope, we enforce this above:

```ts
// Enforce singleton.
if (MySky.instance) {
  return MySky.instance;
}
```

> implement a MySky.getInstance() method which does that and is the recommended way of getting the mysky instance. It calls the constructor if there is no singleton yet or returns the singleton otherwise.

I don't see the advantage of this over the existing entry point `loadMySky`. It seems to me that `getInstance` would just be a wrapper over `loadMySky`. Let me know if I'm missing something though, I'm not familiar with the best practices around singletons.

Reply:
For what it's worth, singletons that accept parameters in the constructor aren't really singletons in the first place. That's why you usually have a static `getInstance()` as ivo says, and actually you usually also make the constructor private. Only ever exposing `getInstance()` which creates the singleton if that hasn't been done yet.

What's the reason you are calling this a singleton in this case?
The way it's created is a bit weird if I'm honest, the `instance` is not 'set' on the class from the outside, it should even be private and get returned by `getInstance()`.

I understand what you're doing though, I'm not sure I'd call it a singleton. If possible I would even drop the `instance` and simply work with a global variable perhaps. You can resolve this comment though if you don't care about all of this and want to keep it the way it is. This is all coming from textbooks so I honestly don't care that much about it. But for someone that does care it will definitely look a bit odd.

Reply:
```
class MySingleton {
     private static instance: MySingleton;
     public static getInstance(): MySingleton {
          if (!this.instance) { 
              this.instance = new MySingleton();
          }
          return this.instance;
     }

     private constructor() {}
}

Reply:
The more I think about it the more I think we should refactor it slightly. Especially the following:
```
if (MySky.instance) {
      return MySky.instance;
}
```
is actually quite dangerous and misleading because it totally neglects the parameters with which the function was called, right?

Reply:
I'm also not really a fan of the `Error` thrown in the MySky constructor. I would make it possible for multiple MySky instances to be created, but structure the code in a way that only one is ever created. That's more easily said than done of course but perhaps in this case the easiest thing to do is use a global variable.

What's the reason only one MySky instance should get created by the way? Perhaps there's no real good reason and we don't have to care about all of this.

Reply:
Created https://github.com/SkynetLabs/skynet-js/issues/374 so I can revisit the singleton pattern and whether it's even needed at a more convenient time, probably involving Karol and Daniel in the discussion.

Reply:
As discussed under https://github.com/SkynetLabs/skynet-js/pull/361#discussion_r804861103, I now agree that moving this to a F/U is a good solution.